pkg/cmd/search/repos/repos.go

```diff
@@ -0,0 +1,198 @@
+	cmdutil.StringEnumFlag(cmd, &opts.Query.Sort, "sort", "", "", []string{"forks", "help-wanted-issues", "stars", "updated"}, "Sort fetched repositories")
+
+	// Query qualifier flags
+	cmdutil.NilBoolFlag(cmd, &opts.Query.Qualifiers.Archived, "archived", "", "Filter based on archive state")
+	cmd.Flags().StringVar(&opts.Query.Qualifiers.Created, "created", "", "Filter based on created at date")
+	cmd.Flags().StringVar(&opts.Query.Qualifiers.Followers, "followers", "", "Filter based on number of followers")
+	cmdutil.StringEnumFlag(cmd, &opts.Query.Qualifiers.Fork, "include-forks", "", "", []string{"false", "true", "only"}, "Include forks in fetched repositories")
+	cmd.Flags().StringVar(&opts.Query.Qualifiers.Forks, "forks", "", "Filter on number of forks")
+	cmd.Flags().StringVar(&opts.Query.Qualifiers.GoodFirstIssues, "good-first-issues", "", "Filter on number of issues with the 'good first issue' label")
+	cmd.Flags().StringVar(&opts.Query.Qualifiers.HelpWantedIssues, "help-wanted-issues", "", "Filter on number of issues with the 'help wanted' label")
+	cmdutil.StringSliceEnumFlag(cmd, &opts.Query.Qualifiers.In, "in", "", nil, []string{"name", "description", "readme"}, "Restrict search to specific part of repository")
+	cmd.Flags().StringSliceVar(&opts.Query.Qualifiers.Language, "language", nil, "Filter based on the coding language")
+	cmd.Flags().StringSliceVar(&opts.Query.Qualifiers.License, "license", nil, "Filter based on license type")
+	cmdutil.NilBoolFlag(cmd, &opts.Query.Qualifiers.Mirror, "mirror", "", "Filter based on mirror state")
+	cmd.Flags().StringVar(&opts.Query.Qualifiers.Org, "org", "", "Filter on organization")
+	cmd.Flags().StringVar(&opts.Query.Qualifiers.Pushed, "updated", "", "Filter on last updated at date")
+	cmd.Flags().StringVar(&opts.Query.Qualifiers.Repo, "repo", "", "Filter on repository name")
+	cmd.Flags().StringVar(&opts.Query.Qualifiers.Size, "size", "", "Filter on a size range, in kilobytes")
+	cmd.Flags().StringVar(&opts.Query.Qualifiers.Stars, "stars", "", "Filter on number of stars")
+	cmd.Flags().StringSliceVar(&opts.Query.Qualifiers.Topic, "topic", nil, "Filter on topic")
+	cmd.Flags().StringVar(&opts.Query.Qualifiers.Topics, "number-topics", "", "Filter on number of topics")
+	cmd.Flags().StringVar(&opts.Query.Qualifiers.User, "user", "", "Filter based on user")
+	cmdutil.StringEnumFlag(cmd, &opts.Query.Qualifiers.Is, "visibility", "", "", []string{"public", "private"}, "Filter based on visibility")
+
+	return cmd
+}
+
+func reposRun(opts *ReposOptions) error {
+	io := opts.IO
+	if opts.Searcher == nil {
+		cfg, err := opts.Config()
+		if err != nil {
+			return err
+		}
+		host, err := cfg.DefaultHost()
+		if err != nil {
+			return err
+		}
+		client, err := opts.HttpClient()
+		if err != nil {
+			return err
+		}
+		opts.Searcher = search.NewSearcher(client, host)
```

Comment:
💅 Could Searcher be always initialized from the outside, i.e. having the Cobra command set up this default Searcher, and tests initializing their own? Then concepts like Config, DefaultHost, and HttpClient wouldn't have to be passed down via `opts`.

Reply:
Do you think that the `Searcher` should be initialized in the default factory like `Browser`? Or were you thinking in `NewCmdRepos`?